census/models.py

```diff
@@ -0,0 +1,107 @@
+from django.db import models
+from django.contrib.sites.models import Site
+from django.contrib.auth.models import User
+from django.db.models import CASCADE
+from ordered_model.models import OrderedModel
+from enum import Enum
+from django.utils import timezone
+from django import urls
+
+
+def now():
+    return timezone.now()
+
+
+class CensusSession(models.Model):
+    census_name = models.TextField(verbose_name='Census name (e.g. 2024 Semester 1)')
+    open_time = models.DateTimeField(verbose_name='Census form opening time')
+    close_time = models.DateTimeField(verbose_name='Census form closing time')
+    response_count = models.IntegerField(verbose_name='Number of responses', default=0)
+    class Meta:
+        ordering = ('-census_name',)
+
+    def census_url(self):
+        try:
+            surl = urls.reverse('census:census-form', kwargs={'session_id': self.id})
+            return 'https://%s%s' % (Site.objects.get_current().domain, surl)
+        except:
+            return 'Save me first'
+    census_url.short_description = 'Census form URL (web address)'
+
+    def census_response_url(self):
+        try:
+            surl = urls.reverse('census:census-results', kwargs={'session_id': self.id})
+            return 'https://%s%s' % (Site.objects.get_current().domain, surl)
+        except:
+            return 'Save me first'
+    census_url.short_description = 'Census result URL (web address)'
+
+    def __str__(self):
+        return ('Census %s' % (self.census_name)).strip()
+
+    def is_census_open(self):
+        return self.open_time <= now() <= self.close_time
+    is_census_open.boolean = True
+
+    def questions(self):
+        return CensusQuestion.objects.filter(session=self).order_by('order')
+
+
+class QuestionType(Enum):
+    LongText = "Long Text"
+    ShortText = "Short Text"
+    SingleChoice = "Single Choice"
+    MultipleChoice = "Multiple Choice"
+
+
+class CensusQuestion(OrderedModel):
+    order_with_respect_to = 'session'
+
+    class Meta(OrderedModel.Meta):
+        pass
+
+    session = models.ForeignKey(CensusSession, on_delete=CASCADE)
+    visible_in_voting = models.BooleanField(verbose_name='Visible for voters', default=True)
+    question_text = models.TextField(verbose_name='Question text', max_length=5000)
+    question_type = models.CharField(verbose_name='Type', max_length=20,
+                                     choices=[(tag.name, tag.value) for tag in QuestionType],
+                                     default=QuestionType.LongText)
+    question_options = models.TextField(verbose_name='Question options (optional)', max_length=300, blank=True)
```

Comment:
Making this only appear/writeable for choice questions could avoid potential confusion. A tool tip/note telling how to make options (I found out quite quickly newlines separate options, but can imagine confusion) might be helpful.